proxy_collection_test.go

```diff
@@ -41,6 +41,30 @@ func TestAddTwoProxiesToCollection(t *testing.T) {
 	}
 }
 
+func TestAddOrReplaceProxyToCollectionWithRandomPort(t *testing.T) {
+	collection := toxiproxy.NewProxyCollection()
+	proxy1 := NewTestProxy("test", "localhost:20000")
+	proxy2 := NewTestProxy("test", "localhost:0")
```

Comment:
As I understand the Listen for NewTestProxy is always "localhost:0". The second argument is upstream.